gnome-image-installer/pages/install/gis-install-page.c

```diff
@@ -116,8 +120,17 @@ gis_install_page_prepare_read (GisPage *page, GError **error)
       g_propagate_error (error, e);
       return FALSE;
     }
-  priv->decompressed = g_converter_input_stream_new (G_INPUT_STREAM (input),
-                                                     G_CONVERTER (priv->decompressor));
+
+  if (priv->decompressor)
+    {
+      priv->decompressed = g_converter_input_stream_new (G_INPUT_STREAM (input),
+                                                         G_CONVERTER (priv->decompressor));
+    }
+  else
+    {
+      priv->decompressed = g_object_ref (input);
+    }
+
   g_object_unref (input);
```

Comment:
I know I wrote this, but - could possibly avoid the ref/unref oddity by just moving the unref up into the if (priv->decompressor) path.